go/vt/vtgate/safe_session.go

```diff
@@ -52,6 +77,41 @@ func NewAutocommitSession(sessn *vtgatepb.Session) *SafeSession {
 	return NewSafeSession(newSession)
 }
 
+// SetAutocommitable sets the state to autocommitable if true.
+// Otherwise, it's notAutocommitable.
+func (session *SafeSession) SetAutocommitable(flag bool) {
+	session.mu.Lock()
+	defer session.mu.Unlock()
+
+	if session.autocommitState == autocommited {
+		panic("BUG: SetAutocommitable: unexpected autocommit state")
+	}
+
+	if flag {
+		session.autocommitState = autocommitable
+	} else {
+		session.autocommitState = notAutocommitable
+	}
+}
+
+// AutocommitToken returns true if we can perform a single round-trip
+// autocommit. If so, the caller is responsible for commiting their
+// transaction.
+func (session *SafeSession) AutocommitToken() bool {
```

Comment:
I would find it clearer if this was simply named `Autocommit()` or `AutocommitIfPossible()`. The use of "Token" suggests that the session will allocate some kind of nonce or other token, not return a boolean and do a state transition.

Reply:
I thought of `CanAutocommit`, or `Autocommittable`. The problem with these names is that they imply that it's a read-only function. In this case, the call causes the session to change state disallowing further transactional operations, and it essentially hands over the commit responsibility to the caller.

Can't think of a better name yet.